core/templates/dev/head/services/exploration-features-backend-api.service.ts

```diff
@@ -17,28 +17,36 @@
  * configured to support.
  */
 
-require('services/services.constants.ajs.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
 
-angular.module('oppia').factory('ExplorationFeaturesBackendApiService', [
-  '$http', 'UrlInterpolationService', 'EXPLORATION_FEATURES_URL',
-  function($http, UrlInterpolationService, EXPLORATION_FEATURES_URL) {
-    return {
-      /**
-       * Retrieves data regarding the features the given exploration supports.
-       *
-       * NOTE: This service requires play-access for the Exploration so that the
-       * features can be fetched in both the exploration player and editor.
-       *
-       * @returns {Object.<string, *>} - Describes the features the given
-       *     exploration supports.
-       */
-      fetchExplorationFeatures: function(explorationId) {
-        return $http.get(
-          UrlInterpolationService.interpolateUrl(
-            EXPLORATION_FEATURES_URL, {exploration_id: explorationId})
-        ).then(function(response) {
-          return response.data;
-        });
-      },
-    };
-  }]);
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { HttpClient } from '@angular/common/http';
```

Comment:
Please move this above. All angular libraries should be together, followed by line break and then Oppia services.

Reply:
Done